tsl/src/data_node_dispatch.c

```diff
@@ -331,6 +332,7 @@ data_node_dispatch_begin(CustomScanState *node, EState *estate, int eflags)
 	sds->flush_threshold = intVal(list_nth(cscan->custom_private, CustomScanPrivateFlushThreshold));
 
 	sds->mcxt = mcxt;
```

Comment:
Maybe you want to call this `stmt_mcxt` to make the usage clear and distinguish it from the batch context.

Reply:
Why do you suggest `stmt_mcxt` specifically. This is the memory context used for executor state, not only the statement.